app/src/main/java/dev/arkbuilders/arkmemo/utils/Utils.kt

```diff
@@ -0,0 +1,67 @@
+package dev.arkbuilders.arkmemo.utils
+
+import android.content.ClipboardManager
+import android.content.Context
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
+import dev.arkbuilders.arkmemo.R
+import dev.arkbuilders.arkmemo.models.SaveNoteResult
+import dev.arkbuilders.arkmemo.ui.views.toast
+import java.nio.file.Files
+import java.nio.file.Path
+import kotlin.io.path.extension
+import kotlin.io.path.forEachLine
+import kotlin.streams.toList
+
+fun Fragment.observeSaveResult(result: LiveData<SaveNoteResult>) {
+    result.observe(this) {
+        if (!isResumed) return@observe
+
+        if (it == SaveNoteResult.SUCCESS) {
+            toast(requireContext(), getString(R.string.ark_memo_note_saved))
+            activity?.onBackPressedDispatcher?.onBackPressed()
+        } else {
+            toast(requireContext(), getString(R.string.ark_memo_note_existing))
```

Comment:
`context` might be `null` here (due to activity is destroyed before toast is shown), which will lead to a runtime crash. Should we handle it more elegantly with above change?

```suggestion
            context?.let { ctx -> toast(ctx, getString(R.string.ark_memo_note_existing)) }
```